plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClient.java

```diff
@@ -352,7 +352,7 @@ public WriteMapping toWriteMapping(ConnectorSession session, Type type)
         }
         if (TIMESTAMP_MILLIS.equals(type)) {
             // TODO use `timestampWriteFunction` (https://github.com/trinodb/trino/issues/6910)
-            return WriteMapping.longMapping("datetime", timestampWriteFunctionUsingSqlTimestamp(TIMESTAMP_MILLIS));
+            return WriteMapping.longMapping("datetime(3)", timestampWriteFunctionUsingSqlTimestamp(TIMESTAMP_MILLIS));
```

Comment:
If I am reading correctly, this would only write `TIMESTAMP(3)` as `datetime(3)` and fallback to legacy write mapping for other types (which doesn't have a mapping for TIMESTAMP type at all so throwing an "Unsupported column type" error).

Let's change this branch to run for anything that is a `TimestampType` and try to see if we can write as `timestamp(p)` where `p` is the precision obtained from the type. See SqlServerClient for reference.

Reply:
That's a relatively straightforward change. I was comparing to `create table` instead of `insert`, but that way makes more sense.

Reply:
On second though, I think this way might be better, especially if we fix #6910 any time soon.

It doesn't really make sense for CTAS to allow more precisions than `INSERT`, so I think we should always write as `DATETIME(3)` until #6910.

That leaves us with the option to automatically convert all `TIMESTAMP`s to precision 3 before inserting on CTAS, but I think that rejecting them is a less surprising behavior, especially if we're soon going to change it to allow other precisions soon.

Reply:
This is definitely correct change. @hashhar note that other timesamps with other precision are simply not supported here and cannot be created at all

Reply:
Thanks for catching this @jirassimok and @findepi . I missed that CTAS flow differs from INSERT.
I'll fix it in #6910 .